conda/fetch.py

```diff
@@ -305,6 +305,11 @@ def fetch_index(channel_urls, use_cache=False, unknown=False, index=None):
             futures = tuple(executor.submit(fetch_repodata, url, use_cache=use_cache,
                                             session=CondaSession()) for url in urls)
             repodatas = [(u, f.result()) for u, f in zip(urls, futures)]
+
+        except Exception as e:
```

Comment:
`except Exception` always sends off warning signals for me.  I'd prefer if we could narrow this down to the actually exception we're trying to catch.  I guess the one from the bug report this is attempting to fix.

also, please add a `log.debug()` here with info about the exception `e`.